port/unix/omrsysinfo.c

```diff
@@ -2799,24 +2800,50 @@ omrsysinfo_get_number_CPUs_by_type(struct OMRPortLibrary *portLibrary, uintptr_t
 			int32_t rc = 0;
 			int64_t cpuQuota = 0;
 			uint64_t cpuPeriod = 0;
-			int32_t numItemsToRead = 1; /* cpu.cfs_quota_us and cpu.cfs_period_us files each contain only one integer value */
 
-			/* If either file read fails, ignore cgroup cpu quota limits and continue */
-			rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_CPU, "cpu.cfs_quota_us", numItemsToRead, "%ld", &cpuQuota);
-			if (0 == rc) {
-				rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_CPU, "cpu.cfs_period_us", numItemsToRead, "%lu", &cpuPeriod);
+			if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE)) {
+				/* cpu.cfs_quota_us and cpu.cfs_period_us files each contain only one integer value. */
+				int32_t numItemsToRead = 1;
+
+				/* If either file read fails, ignore cgroup cpu quota limits and continue. */
+				rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_CPU, "cpu.cfs_quota_us", numItemsToRead, "%ld", &cpuQuota);
 				if (0 == rc) {
-					int32_t numCpusQuota = (int32_t) (((double) cpuQuota / cpuPeriod) + 0.5);
+					rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_CPU, "cpu.cfs_period_us", numItemsToRead, "%lu", &cpuPeriod);
+				}
+			} else {
+				/* Cgroup v2 is available. Read cpu.max file which contains the quota and period
+				 * in the format "$QUOTA $PERIOD". The value "max" for $QUOTA indicates no limit.
+				 */
+				int32_t numItemsToRead = 2;
```

Comment:
The `numItemsToRead` is not always guaranteed to be `true`. There can also be one number written in `cpu.max`. With `numItemsToRead == 2`, `readCgroupSubsystemFile:(numItemsToRead != rc)` will fail for the one number scenario. We need to clearly state that cgroup cpu quota limits will be ignored for this scenario. Add a detailed comment.

Ref: https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html

> `$MAX $PERIOD` which indicates that the group may consume upto $MAX in each $PERIOD duration. “max” for $MAX indicates no limit. If only one number is written, $MAX is updated.

Reply:
I think there will always be two values in cpu.max. In the ref, it states
> A read-write two value file which exists on non-root cgroups.

Also when I try to write one number only, it changes $MAX but $PERIOD remains in the file.
```
root@purpurin1 ~/d/o/omr (cpuQuota)# cat /sys/fs/cgroup/user.slice/cpu.max
max 100000
root@purpurin1 ~/d/o/omr (cpuQuota)# echo 100000 > /sys/fs/cgroup/user.slice/cpu.max
root@purpurin1 ~/d/o/omr (cpuQuota)# cat /sys/fs/cgroup/user.slice/cpu.max
100000 100000
```